src/TopoMojo.Hypervisor/Proxmox/ProxmoxVlanManager.cs

```diff
@@ -198,7 +273,14 @@ private async Task<IEnumerable<PveVnetOperationResult>> DebounceVnetOperations(I
 
                 if (vnetsToDelete.Any())
                 {
-                    var vnetNamesToDelete = vnetsToDelete.Select(n => _nameService.ToPveName(n.NetworkName));
+                    // Only delete vnets that haven't been created recently to avoid accidentally deleting
+                    // a vnet that is in use. Also, don't delete a vnet if we created it in this batch
+                    var vnetNamesToDelete = vnetsToDelete
```

Comment:
Wanted to ask a little about this. Did you have problems with stuff being deleted incorrectly? At least for the case of not wanting to delete something we created in the same batch, we could instead inspect the `results` collection during deletes to make sure that the ones being deleted don't appear in results from created. Were dupe calls happening? I'm just making sure I understand the benefit of maintaining this cache's state, basically.

Reply:
This is because I removed deleting networks when calling pod.DeleteAll, to match what the vmware side is doing. There, network deletes only happen during a periodic cleanup call. So if you destroy and redeploy the vms in a gamespace multiple times, it just reuses the existing vnets after the first and saves time for the user.

The problem is in the unlikely event that the cleanup tries to delete a vnet just as you are redeploying a vm that is going to use it. So the cache ensures that we only delete vnets if they haven't been used in a while and since all vnet creation goes through this lock, I think that should be enough to ensure that situation never happens.